src/ion/moves.rs

```diff
@@ -228,6 +226,7 @@ impl<'a, F: Function> Env<'a, F> {
                     );
                 }
 
+                // TODO: Is this even possible to trigger without prog-moves?
```

Comment:
I'm not completely sure about this part. By my understanding, it shouldn't be possible to reach these conditions without progmoves but I'm not 100% sure.

Reply:
This condition is triggered whenever there is a split, actually -- e.g., if we're forced to move a value to a different register prior to an instruction because another higher-priority liverange def'd by that instruction gets that register, then the split could happen at an arbitrary point within the block.

(FWIW, the question could also be answered by fuzzing -- I suspect if you add a panic here you'll see it hit shortly, even given that fuzzing has never generated functions with program-moves.)